### PR TITLE
Rename context.trace -> context.span

### LINKED ISSUE #544: Rename context.trace -> context.span
It confuses people since it's not exclusively related to tracing, but rather the whole observability system.

### PR DESCRIPTION
**Pull request by spladug**
The span is for the whole observer system, not just for tracing. Even
though the span concept comes from tracing.

Fixes #544

### CODE REVIEW

File: baseplate/__init__.py [hunk 1]
```diff
@@ -704,7 +704,7 @@ def make_child(
                 context_copy,
                 self.baseplate,
             )
-        context_copy.trace = span
+        context_copy.span = span
```
Comment (MelissaCole) [inline baseplate/__init__.py:707]: Do you want to update the comment on L718?
> during requests as the ``trace`` attribute.

**Comment by MelissaCole** [comment]
:fish: lgtm!

File: baseplate/__init__.py (hunk 1 above)
Comment (spladug) [reply baseplate/__init__.py:707]: thanks!

**Review by kemitche — APPROVED** [review]
Thanks for this!

**Review by bradengroom — APPROVED** [review]

**Pull request merged**